src/Generator/AwinCOM.php

```diff
@@ -44,126 +64,274 @@ public function __construct(ArrayHelper $arrayHelper)
+                'manufacturer'          => $manufacturer,
+                'prod_url'              => $this->elasticExportHelper->getMutatedUrl($variation, $settings, true, false),
+                'prod_ean'              => $this->elasticExportHelper->getBarcodeByType($variation, $settings->get('barcode')),
+                'shipping_costs'        => $shippingCost,
+                'base_price'            => $basePrice,
+                'base_price_amount'     => $basePriceList['lot'],
+                'base_price_unit'       => $basePriceList['unit'],
+            ];
+
+            $this->addCSVContent(array_values($data));
+
+            $this->getLogger(__METHOD__)->debug('ElasticExportAwinCOM::log.variationConstructRowFinished', [
+                'Data row duration' => 'Row printing took: ' . (microtime(true) - $rowTime),
+            ]);
+        }
+        else
+        {
+            $this->getLogger(__METHOD__)->info('ElasticExportAwinCOM::log.variationNotPartOfExportPrice', [
+                'VariationId' => (string)$variation['id']
+            ]);
         }
     }
 
     /**
-     * Creates an array with the rest of data needed from the ItemDataLayer.
+     * Get the shipping cost.
      *
-     * @param RecordList $idlResultList
+     * @param $variation
+     * @return string
      */
-    private function createIdlArray($idlResultList)
+    private function getShippingCost($variation):string
     {
-        if($idlResultList instanceof RecordList)
+        $shippingCost = null;
+        if(isset($this->shippingCostCache) && array_key_exists($variation['data']['item']['id'], $this->shippingCostCache))
         {
-            foreach($idlResultList as $idlVariation)
-            {
-                if($idlVariation instanceof Record)
-                {
-                    $this->idlVariations[$idlVariation->variationBase->id] = [
-                        'itemBase.id' => $idlVariation->itemBase->id,
-                        'variationBase.id' => $idlVariation->variationBase->id,
-                        'variationRetailPrice.price' => $idlVariation->variationRetailPrice->price,
-                        'variationRetailPrice.currency' => $idlVariation->variationRetailPrice->currency
-                    ];
-                }
-            }
+            $shippingCost = $this->shippingCostCache[$variation['data']['item']['id']];
+        }
+
+        if(!is_null($shippingCost) && $shippingCost != '0.00')
+        {
+            return $shippingCost;
+        }
+
+        return '';
+    }
+
+    /**
+     * Get the manufacturer name.
+     *
+     * @param $variation
+     * @return string
+     */
+    private function getManufacturer($variation):string
+    {
+        if(isset($this->manufacturerCache) && array_key_exists($variation['data']['item']['id'], $this->manufacturerCache))
+        {
+            return $this->manufacturerCache[$variation['data']['item']['id']];
```

Comment:
Maybe its better if we select the manufacturer one time and keep it in cache for other items. If its possible, we could separate the manufacturer by manufacturer id, if we have that id.